include/eosio/to_key.hpp

```diff
@@ -260,58 +254,45 @@ UInt float_to_key(T value) {
 }
 
 template <typename T, typename S>
-result<void> to_key(const T& obj, S& stream) {
+void to_key(const T& obj, S& stream) {
    if constexpr (std::is_floating_point_v<T>) {
       if constexpr (sizeof(T) == 4) {
-         return to_key(float_to_key<uint32_t>(obj), stream);
+         to_key(float_to_key<uint32_t>(obj), stream);
       } else {
          static_assert(sizeof(T) == 8, "Unknown floating point type");
-         return to_key(float_to_key<uint64_t>(obj), stream);
+         to_key(float_to_key<uint64_t>(obj), stream);
       }
    } else if constexpr (std::is_integral_v<T>) {
       auto v = static_cast<std::make_unsigned_t<T>>(obj);
       v -= static_cast<std::make_unsigned_t<T>>(std::numeric_limits<T>::min());
       std::reverse(reinterpret_cast<char*>(&v), reinterpret_cast<char*>(&v + 1));
-      return stream.write_raw(v);
+      stream.write_raw(v);
    } else if constexpr (std::is_enum_v<T>) {
       static_assert(!std::is_convertible_v<T, std::underlying_type_t<T>>, "Serializing unscoped enum");
-      return to_key(static_cast<std::underlying_type_t<T>>(obj), stream);
+      to_key(static_cast<std::underlying_type_t<T>>(obj), stream);
    } else {
-      result<void> r = outcome::success();
       eosio::for_each_field(obj, [&](const auto& member) {
-         if (r)
-            r = to_key(member, stream);
+         to_key(member, stream);
       });
-      return r;
    }
 }
 
 template <typename T>
-result<void> convert_to_key(const T& t, std::vector<char>& bin) {
+void convert_to_key(const T& t, std::vector<char>& bin) {
    size_stream ss;
-   auto        r = to_key(t, ss);
-   if (!r)
-      return r.error();
+   to_key(t, ss);
    auto orig_size = bin.size();
    bin.resize(orig_size + ss.size);
    fixed_buf_stream fbs(bin.data() + orig_size, ss.size);
-   r = to_key(t, fbs);
-   if (!r)
-      return r.error();
-   if (fbs.pos == fbs.end)
-      return outcome::success();
-   else
-      return stream_error::underrun;
+   to_key(t, fbs);
+   check( fbs.pos == fbs.end, convert_stream_error(stream_error::underrun) );
 }
 
 template <typename T>
-result<std::vector<char>> convert_to_key(const T& t) {
+std::vector<char> convert_to_key(const T& t) {
    std::vector<char> result;
-   auto              r = convert_to_key(t, result);
-   if (r)
-      return std::move(result);
-   else
-      return r.error();
+   convert_to_key(t, result);
+   return std::move(result);
```

Comment:
You removed the unneeded `std::move` of result other places, but left it here.

Reply:
I found the others because the compiler was warning about them.  I wonder why I'm not seeing a warning here.